provisioner/puppet-masterless/provisioner.go

```diff
@@ -227,8 +275,15 @@ func (p *Provisioner) Provision(ui packer.Ui, comm packer.Communicator) error {
 	}
 
 	// Execute Puppet
+	var facterVarsString string
+	if p.config.GuestOSType == provisioner.UnixOSType {
+		facterVarsString = strings.Join(facterVars, " ")
+	} else {
+		facterVarsString = "SET " + strings.Join(facterVars, " && SET ")
+	}
+
 	p.config.ctx.Data = &ExecuteTemplate{
```

Comment:
The code for ModulePath will only work on Linux. Windows needs the ";" and not ":" like a Linux machine. I tested locally and put the code to be  strings.Join(modulePaths, p.guestOSTypeConfig.separator and put the seperator field in the  guestOSTypeConfig struct